src/themes/material/assets/material.js

```diff
@@ -35,3 +36,70 @@ $('a[href^="#"]:not(a[href$="!"])').click(function() {
 
     return false;
 });
+
+/**
+ * Handles closing sidenav when keyboard focus leaves it and escape key functionality
+ */
+function initSidenavAccessibility() {
+    // Wait for Materialize to initialize
+    setTimeout(function() {
+        var sidenavInstance = M.Sidenav.getInstance(document.getElementById('nav-mobile'));
+        var sidenavTrigger = document.querySelector('.sidenav-trigger');
+        var sidenavElement = document.getElementById('nav-mobile');
+        
+        if (!sidenavInstance || !sidenavElement) {
+            return;
+        }
+        
+        var originalTrigger = null;
+        
+        // Handle escape key to close sidenav
+        document.addEventListener('keydown', function(e) {
+            if (e.key === 'Escape' && sidenavInstance.isOpen) {
+                sidenavInstance.close();
+                // Return focus to the trigger button
+                if (sidenavTrigger) {
+                    sidenavTrigger.focus();
+                }
+            }
+        });
+        
+        // Handle focus management when sidenav opens
+        sidenavElement.addEventListener('focusin', function(e) {
+            // Store the trigger when sidenav opens
+            if (!originalTrigger && sidenavTrigger) {
+                originalTrigger = sidenavTrigger;
+            }
+        });
+        
+        // Handle focus leaving the sidenav
+        sidenavElement.addEventListener('focusout', function(e) {
+            setTimeout(function() {
+                var activeElement = document.activeElement;
+                var isFocusInSidenav = sidenavElement.contains(activeElement) || 
+                                     sidenavElement === activeElement;
+                
+                // If focus is not in sidenav and sidenav is open, close it
+                if (!isFocusInSidenav && sidenavInstance.isOpen) {
+                    sidenavInstance.close();
+                    if (originalTrigger && originalTrigger.offsetParent !== null) {
+                        originalTrigger.focus();
+                    }
```

Comment:
I'm a bit out of my depth here, so I thought I'd ask. Why is it important to close the sidenav when focus changes out of it?